source/AI.cpp

```diff
@@ -3652,9 +3652,9 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			autoPilot = activeCommands;
 	}
 	bool shouldAutoAim = false;
-	if(Preferences::Has("Automatic aiming") && !command.Turn() && !ship.IsBoarding()
-			&& ((target && target->GetSystem() == ship.GetSystem() && target->IsTargetable())
-				|| ship.GetTargetAsteroid())
+	if((Preferences::AutoAimIndex() ? ((Preferences::AutoAimIndex() == 1) ? (activeCommands.Has(Command::PRIMARY)) : true) : false)
```

Comment:
Why only for primary weapons? I would expect "when firing" to also apply to secondary weapons.

Reply:
This was the old behavior that only firing primary weapons would activate auto aim, so I guess this PR is just restoring old behavior using a preference. The reason the preference behavior was changed though was exactly because secondary weapons didn't make use of auto aim, so we should address that if we're reintroducing the "only when firing" behavior.

Reply:
It applies to secondaries as well now